flink-core/src/main/java/org/apache/flink/api/common/state/StateTtlConfig.java

```diff
@@ -204,16 +218,31 @@ public Builder neverReturnExpired() {
 		 * Sets the time characteristic.
 		 *
 		 * @param timeCharacteristic The time characteristic configures time scale to use for ttl.
+		 *
+		 * @deprecated will be removed in a future version in favor of {@link #setTtlTimeCharacteristic}
 		 */
+		@Deprecated
 		@Nonnull
 		public Builder setTimeCharacteristic(@Nonnull TimeCharacteristic timeCharacteristic) {
-			this.timeCharacteristic = timeCharacteristic;
+			// We only support ProcessingTime now, so transfer to  {@link TtlTimeCharacteristic.ProcessingTime} directly.
```

Comment:
What's happening here? We're silently ignoring the parameter that was passed in?

Reply:
@aljoscha Currently, `TimeCharacteristic` contains `ProcessingTime` only, we can assert that the parameter passed in will always be `TimeCharacteristic#ProcessingTime`, so transfer to `TtlTimeCharacteristic.ProcessingTime` directly.

Reply:
`TimeCharacteristic` is also deprecated and we will not add other time types to it to encourage `TtlTimeCharacteristic` usage.

Reply:
I see, we should put in a check to make sure that we don't break this in the future.